### PR TITLE
Update Chinese word segmentation and score calculation and sessionId

### PR DESCRIPTION
**Pull request by jinshan1979**
## Purpose

* update Chinese word segmentation 
* update score calculation
* adding sessionId
* update for csharp/cpp/java language

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: samples/csharp/sharedcontent/console/speech_recognition_samples.cs [hunk 1]
```diff
@@ -1185,37 +1200,111 @@ public static async Task PronunciationAssessmentConfiguredWithJson()
             }
         }
 
+        public static List<string> GetReferenceWords(string waveFilename, string referenceText, string language, SpeechConfig speechConfig)
+        {
+            var audioConfig = AudioConfig.FromWavFileInput(waveFilename);
+            //var speechConfig = SpeechConfig.FromSubscription(speechKey, serviceRegion);
```
Comment (wangkenpu) [inline samples/csharp/sharedcontent/console/speech_recognition_samples.cs:1206]: Can you remove this line?

File: samples/csharp/sharedcontent/console/speech_recognition_samples.cs [hunk 2]
```diff
@@ -1263,13 +1354,25 @@ public static async Task PronunciationAssessmentContinuousWithFile()
...
                     // even if miscue is enabled.
                     // We need to compare with the reference text after received all recognized words to get these error words.
-                    string[] referenceWords = referenceText.ToLower().Split(' ');
-                    for (int j = 0; j < referenceWords.Length; j++)
+                    string[] referenceWords;
+
+                    if (language == "zh-CN")
                     {
-                        referenceWords[j] = Regex.Replace(referenceWords[j], "^[\\p{P}\\s]+|[\\p{P}\\s]+$", "");
+                        referenceWords = GetReferenceWords(waveFileName, referenceText, language, config).ToArray();
```
Comment (wangkenpu) [inline samples/csharp/sharedcontent/console/speech_recognition_samples.cs:1367]: here should use a very short audio like Yan did in the python sample code.

File: samples/java/jre/console/src/com/microsoft/cognitiveservices/speech/samples/console/SpeechRecognitionSamples.java [hunk 3]
```diff
@@ -1367,9 +1450,17 @@ else if (e.getResult().getReason() == ResultReason.NoMatch) {
             // For continuous pronunciation assessment mode, the service won't return the words with `Insertion` or `Omission`
             // even if miscue is enabled.
             // We need to compare with the reference text after received all recognized words to get these error words.
-            String[] referenceWords = referenceText.toLowerCase().split(" ");
-            for (int j = 0; j < referenceWords.length; j++) {
-                referenceWords[j] = referenceWords[j].replaceAll("^\\p{Punct}+|\\p{Punct}+$","");
+            String[] referenceWords;
+            if (lang == "zh-CN")
+            {
+                referenceWords = getReferenceWords(wavFileName, referenceText, lang, config).toArray(new String[0]);
```
Comment (wangkenpu) [inline samples/java/jre/console/src/com/microsoft/cognitiveservices/speech/samples/console/SpeechRecognitionSamples.java:1456]: The same as C# code.

**Review by wangkenpu — CHANGES_REQUESTED** [review]

File: samples/csharp/sharedcontent/console/speech_recognition_samples.cs (hunk 1 above)
Comment (jinshan1979) [reply samples/csharp/sharedcontent/console/speech_recognition_samples.cs]: removed

File: samples/csharp/sharedcontent/console/speech_recognition_samples.cs (hunk 2 above)
Comment (jinshan1979) [reply samples/csharp/sharedcontent/console/speech_recognition_samples.cs]: done, update with short audio

File: samples/java/jre/console/src/com/microsoft/cognitiveservices/speech/samples/console/SpeechRecognitionSamples.java (hunk 3 above)
Comment (jinshan1979) [reply samples/java/jre/console/src/com/microsoft/cognitiveservices/speech/samples/console/SpeechRecognitionSamples.java]: done the update

**Review by wangkenpu — DISMISSED** [review]

**Review by yaden-ms — APPROVED** [review]

**Review by wangkenpu — APPROVED** [review]

**Pull request merged**